torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -176,18 +176,18 @@ static inline Variable applySlicing(
     variable_list& outIndices,
     bool is_tracing,
     const at::Device& self_device,
-    const c10::optional<IntArrayRef>& self_sizes,
+    const c10::optional<int64_t>& self_ndim,
```

Comment:
This looks totally different

Reply:
Well well well looks like this is way better

Reply:
It is.
But the only thing this function is doing is checking that the number of dimension is correct. So I changed this altogether so that we never have to call sym_sizes()